java/src/hu/holyoil/controller/SunController.java

```diff
@@ -38,18 +39,34 @@ public String toString() {
      * Pseudo random szám, ami a következő napviharig hátralévő köröket számolja
      * <p>körönként csökken,napvihar után újra sorsolódik</p>
      */
-    private Integer turnsUntilNextSunstorm;
+    private int turnsUntilNextSunstorm;
     /**
      * A pályán található összes aszteroidát tartalmazó lista
      */
     private List<Asteroid> asteroids;
 
+    /**
+     * Újraindítja a visszaszámláló turnsUntilNextSunstorm számlálót egy random számra 20 és 50 között
+     */
+    private void RestartCountdown() {
+        Random random = new Random();
+        turnsUntilNextSunstorm = random.nextInt(50 - 20 + 1) + 20;
+    }
+
     /**
      * minden körben végrehajt egy lépést
      */
     @Override
     public void Step() {
-        System.out.println("Stepping");
+        Logger.Log(this, "Steps" + (Main.isRandomEnabled ? "" : " - switched off!"));
+
+        if (Main.isRandomEnabled) {
```

Comment:
This does not change when randomness is disabled

Reply:
Done